BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -60,6 +61,39 @@
 
   private ExportBibTex exportBibTex;
   private ImportBibTex importBibTex;
+  ActivityResultLauncher<Intent> filePickerActivityResultLauncher = registerForActivityResult(
+      new ActivityResultContracts.StartActivityForResult(),
+      new ActivityResultCallback<ActivityResult>() {
+        @Override
+        public void onActivityResult(ActivityResult result) {
+          if (result.getResultCode() == Activity.RESULT_OK) {
+            Intent data = result.getData();
+
+            if (data != null) {
+
+              Uri uri = data.getData();
+              if (importBibTex.isBibFile(UriUtils.getFullUriPath(context, uri))) {
+
+                handleImport(uri);
+
+              } else {
+                showDialogNonBibFile();
+              }
+            }
+          }
+        }
+      });
+  private final ActivityResultLauncher<String> requestPermissionLauncher =
+      registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
+        if (isGranted) {
+          filePicker();
+
+
+        } else {
+          Toast.makeText(getContext(), R.string.storage_permission_denied,
+              Toast.LENGTH_SHORT).show();
+        }
+      });
```

Comment:
Please add a newline.
```suggestion
      });
      
```